src/resmom_win/start_exec.c

```diff
@@ -2253,6 +2400,13 @@ finish_exec(job *pjob)
 					free_attrlist(&argv_list);
 					free_str_array(hook_output.env);
 					free_str_array(argv_out);
+					if (do_tolerate_node_failures(pjob)) {
+						vnl_free(vnl_fails);
+						vnl_fails = NULL;
+						vnl_free(vnl_good);
+						vnl_good = NULL;
+						free(old_exec_vnode);
+					}
```

Comment:
Another repeated block of code that might work better as a macro.

Reply:
I've macro-ed this.